### PR TITLE
Devenv: Add tempo data source

### PR DESCRIPTION
**Pull request by aocenas**
Adds a block for tempo datasource.

I also added some readme for how to setup some logging and tracing in Loki so it is easier to setup data links. Issue is this needs manual change of the docker-compose which I usually do and manually run docker-compose rather than use the `make devenv`. As you may need the links setup differently because of usecase I am not sure if it make sense to hardcode them, also logging driver in loki for example needs a docker plugin installed so requiring that if you do not need it may be an overkill.

### CODE REVIEW

**Comment by zoltanbedi** [comment]
The tempo block returns with an error:
```
failed parsing config: failed to parse configFile /etc/tempo.yaml: yaml: unmarshal errors:

line 23: field max_block_bytes not found in type ingester.Config

line 36: field block not found in type tempodb.Config
```
Did I have to change something in the config in order to make this work?

File: devenv/docker/blocks/loki/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+By default this block is setup to scrape logs from /var/logs. If you need to log some service from the docker-compse you can add:
+```
+    # For this to work you need to install the logging driver see https://github.com/grafana/loki/tree/master/cmd/docker-driver#plugin-installation
+    logging:
+      driver: loki
+      options:
+        loki-url: "http://loki:3100/loki/api/v1/push"
+```
+
```
Comment (ivanahuckova) [inline devenv/docker/blocks/loki/README.md:9]: I am not sure if I am doing something wrong, but I am not getting any logs with **traceId**. 
Is my config wrong? Or anything else? Also, which logs are going to have traceId? 
``` 
# datasource URL: http://localhost:3100/
  loki:
    image: grafana/loki:master
    ports:
      - "3100:3100"
    command: -config.file=/etc/loki/local-config.yaml
    logging:
      driver: loki
      options:
        loki-url: "http://loki:3100/loki/api/v1/push" 
    environment:
      - JAEGER_AGENT_HOST=tempo
      - JAEGER_AGENT_PORT=6831
      - JAEGER_SAMPLER_TYPE=const
      - JAEGER_SAMPLER_PARAM=1
  promtail:
    image: grafana/promtail:master
    volumes:
      - ./docker/blocks/loki/config.yaml:/etc/promtail/docker-config.yaml
      - /var/log:/var/log
      - ../data/log:/var/log/grafana
    command:
      -config.file=/etc/promtail/docker-config.yaml
```

**Review by ivanahuckova — COMMENTED** [review]
Also, related to this PR and our conversation with @aocenas, I have created this issue: https://github.com/grafana/grafana/issues/31222

**Comment by ivanahuckova** [comment]
Also, one more thing - could you add `gdev-tempo` data source?

File: devenv/docker/blocks/tempo/tempo.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    max_compaction_objects: 1000000    # maximum size of compacted blocks
+    block_retention: 1h
+    compacted_block_retention: 10m
+
+storage:
+  trace:
+    backend: local                     # backend configuration to use
+    block:
+      bloom_filter_false_positive: .05 # bloom filter false positive rate.  lower values create larger filters but fewer false positives
+      index_downsample: 10             # number of traces per index record
```
[the remark this answers is not in the document]
Comment (aocenas) [reply devenv/docker/blocks/tempo/tempo.yaml:38]: Yeah seems like some of the options got removed in later versions of tempo

File: devenv/docker/blocks/loki/README.md (hunk 1 above)
Comment (aocenas) [reply devenv/docker/blocks/loki/README.md:9]: I think that should work but only query logs get traceId. So I have to run a query first (the same to get the logs) and then I get a log like this:
Screenshot from 2021-03-08 11-29-10

File: devenv/docker/blocks/tempo/docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+  tempo:
+    image: grafana/tempo:latest
+    command: ["-config.file=/etc/tempo.yaml"]
+    ports:
+      - "6831:6831"
+      - "3101:3101"
+    volumes:
+      - ./docker/blocks/tempo/tempo.yaml:/etc/tempo.yaml
+
+  tempo-query:
```
Comment (zoltanbedi) [inline devenv/docker/blocks/tempo/docker-compose.yaml:10]: Do we still want to add tempo-query?

**Comment by aocenas** [comment]
I think/hope this should be working now :D

**Review by zoltanbedi — APPROVED** [review]
lgtm 🍤

**Pull request merged**